applications/satellite-imagery-geotiff.ipynb

```diff
@@ -0,0 +1,334 @@
+   "metadata": {},
+   "source": [
+    "The image has separate blocks for each band with block size 512 x 512. \n",
+    "\n",
+    "## Create XArray datasets"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import xarray as xr\n",
+    "red = xr.open_rasterio('red.tif', chunks={'band': 1, 'x': 1024, 'y': 1024})\n",
+    "nir = xr.open_rasterio('nir.tif', chunks={'band': 1, 'x': 1024, 'y': 1024})\n",
+    "nir"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Each dataset's data is a Dask array."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "red.variable.data"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Optional: create a Dask client\n",
+    "\n",
+    "You can start a Dask client to monitor execution with the dashboard."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import dask\n",
+    "from dask.distributed import Client\n",
+    "client = Client(processes=False)\n",
+    "client"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Rescale bands using Landsat metadata\n",
+    "\n",
+    "The Landsat Level 1 images are delivered in a quantized format. This has to be [converted to top-of-atmosphere reflectance](https://landsat.usgs.gov/using-usgs-landsat-8-product) using the provided metadata.\n",
+    "\n",
+    "First we define convenience functions to load the rescaling factors and transform a dataset. The red band is band 4 and near infrared is band 5."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def load_scale_factors(filename, band_number):\n",
+    "    with open(filename) as f:\n",
+    "        metadata = json.load(f)\n",
+    "    M_p = metadata['L1_METADATA_FILE'] \\\n",
+    "                  ['RADIOMETRIC_RESCALING'] \\\n",
+    "                  ['REFLECTANCE_MULT_BAND_{}'.format(band_number)]\n",
+    "    A_p = metadata['L1_METADATA_FILE'] \\\n",
+    "                  ['RADIOMETRIC_RESCALING'] \\\n",
+    "                  ['REFLECTANCE_ADD_BAND_{}'.format(band_number)]\n",
+    "    return M_p, A_p"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def calculate_reflectance(ds, band_number, metafile='meta.json'):\n",
+    "    M_p, A_p = load_scale_factors(metafile, band_number)\n",
+    "    toa = M_p * ds + A_p\n",
+    "    return toa"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "red = calculate_reflectance(red, band_number=4)\n",
+    "nir = calculate_reflectance(nir, band_number=5)"
```

Comment:
Slightly confusing to overwrite original meaning of `red` and `nir`. Might be better to use new variable names like `red_toa` or `nir_toa`.

Reply:
I agree, fixed